OMCompiler/Compiler/SimCode/SimCodeMain.mo

```diff
@@ -1005,116 +1008,157 @@ public function translateModel "
   input FCore.Graph inEnv;
   input Absyn.Path className "path for the model";
   input String inFileNamePrefix;
-  input Boolean addDummy "if true, add a dummy state";
+  input Boolean runBackend "if true, run the backend as well. This will run SimCode and Codegen as well.";
+  input Boolean runSilent "if true, flat modelica code will not be dumped to out stream";
   input Option<SimCode.SimulationSettings> inSimSettingsOpt;
   input Absyn.FunctionArgs args=Absyn.emptyFunctionArgs "labels for remove terms";
-  output list<String> outStringLst;
+  output list<String> outLibs;
   output String outFileDir;
   output list<tuple<String, Values.Value>> resultValues;
 protected
   FCore.Cache inCache = cache;
+  Real timeFrontend=0.0;
+  DAE.DAElist dae;
+  FCore.Graph env;
+  Option<DAE.DAElist> odae;
+  DAE.FunctionTree funcs;
+  list<Option<Integer>> allRoots;
+  FlatModel flatModel;
+  FunctionTree funcTree;
+  NBackendDAE bdae;
+  Boolean dumpValidFlatModelicaNF;
+  String flatString = "", NFFlatString = "";
+
+algorithm
+  FlagsUtil.setConfigBool(Flags.BUILDING_MODEL, true);
+
+  outLibs := {};
+  outFileDir := "";
+  resultValues := {};
+
+  dumpValidFlatModelicaNF := not runSilent and Config.flatModelica();
+
+  // new backend - also activates new frontend by default
+  if Flags.getConfigBool(Flags.NEW_BACKEND) then
+    // ToDo: set permanently matching -> SBGraphs
+    System.realtimeTick(ClockIndexes.RT_CLOCK_FRONTEND);
+    ExecStat.execStatReset();
+
+    (flatModel, funcTree, NFFlatString) := CevalScriptBackend.runFrontEndWorkNF(className, false, dumpValidFlatModelicaNF);
+    timeFrontend := System.realtimeTock(ClockIndexes.RT_CLOCK_FRONTEND);
+    ExecStat.execStat("FrontEnd");
+
+    if dumpValidFlatModelicaNF then
+      flatString := NFFlatString;
+    elseif not runSilent then
+      (dae, funcs) := NFConvertDAE.convert(flatModel, funcTree);
+      flatString := DAEDump.dumpStr(dae, funcs);
```

Comment:
For the new backend we could possibly use `FlatModel.toString` instead and avoid converting to DAE, since I assume we don't have any test cases that check the flat output with the new backend anyway. But it probably doesn't matter much.

Reply:
> since I assume we don't have any test cases that check the flat output with the new backend anyway

Probably not. We can try it. 

However, I think it is better to do that where the NF create the valid flat Modelica string now. That is, the NF should generate some flat output based on the options it is given and return either the valid or normal flat Modelica code. Somewhere around [here](https://github.com/OpenModelica/OpenModelica/blob/master/OMCompiler/Compiler/NFFrontEnd/NFInstUtil.mo#L72-L82) if possible. Then this can be changed to check if there is some output string from the NF and print that. OR even better the NF itself can dump it, although that can perhaps require a bit more adjustment in other places.